pkg/controller/operator/operator_controller.go

```diff
@@ -0,0 +1,163 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("operator-controller")
+
+// Add creates a new Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	k8sClient := mgr.GetClient()
+	// Need to pre-load the config ConfigMap to avoid locking/caching issues when
+	// 2 or more controllers start simultaneaously and try to load the same object
+	dedicatedadmin.GetOperatorConfig(context.Background(), k8sClient)
+
+	return &ReconcileNamespace{client: k8sClient, scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("operator-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Namespace
+	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileNamespace{}
+
+// ReconcileNamespace reconciles a Namespace obj
+type ReconcileNamespace struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster Namespace objects and create required operator resources
+func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.Background()
+	// Initialize logging obj
+	reqLogger := log.WithValues("Request.Namespace", request.Name)
+
+	// Check if the namespace is black listed - administrative namespaces where we
+	// don't want to add the dedicated-admin rolebinding, e. g kube-system, openshift-logging
+	if request.Name != operatorconfig.OperatorNamespace {
+		reqLogger.Info("Not operator namespace - Skipping")
+
+		return reconcile.Result{}, nil
+	}
+
+	// Get the Namespace instance
+	ns := &corev1.Namespace{}
+	err := r.client.Get(ctx, request.NamespacedName, ns)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, it can be transitioning to the final desired state
+			// e. g. deletion or creation still in progress. Return and retry again
+			reqLogger.Info("Object not ready")
+			return reconcile.Result{}, err
+		}
+		// Error reading the obj
+		reqLogger.Info("Error Getting Namespace")
+		return reconcile.Result{}, err
+	}
+
+	// Namespace is being deleted
+	if ns.Status.Phase == corev1.NamespaceTerminating {
```

Comment:
this block is obsolete, since this is the namespace where the operator is running, being impossible for it to be deleted without terminating the operator itself